requestmanager/executor/executor_test.go

```diff
@@ -292,9 +298,62 @@ type requestExecutionEnv struct {
 
 	// deps
 	tbc *testutil.TestBlockChain
-	fal *testloader.FakeAsyncLoader
 }
 
+type fakeReconciledLoader struct {
+	responsesLk sync.Mutex
+	responses   map[datamodel.Link]chan types.AsyncLoadResult
+	lastLoad    datamodel.Link
+	online      bool
+	cb          func(datamodel.Link)
+}
+
+func (frl *fakeReconciledLoader) onAsyncLoad(cb func(datamodel.Link)) {
+	frl.cb = cb
+}
+
+func (frl *fakeReconciledLoader) responseOn(link datamodel.Link, result types.AsyncLoadResult) {
+	response := frl.asyncLoad(link, true)
+	response <- result
+	close(response)
+}
+
+func (frl *fakeReconciledLoader) successResponseOn(blks []blocks.Block) {
+
+	for _, block := range blks {
+		frl.responseOn(cidlink.Link{Cid: block.Cid()}, types.AsyncLoadResult{Data: block.RawData(), Local: false, Err: nil})
+	}
+}
+
+func (frl *fakeReconciledLoader) asyncLoad(link datamodel.Link, force bool) chan types.AsyncLoadResult {
+	frl.responsesLk.Lock()
+	response, ok := frl.responses[link]
+	if !ok || force {
```

Comment:
should this be an `&&` so we don't overwrite with a new one if it happens to exist? or do you think it doesn't matter for the purpose of a fake?

Reply:
the idea of force is to force the creation of a new channel, so it should be an ||